addons/sale_timesheet/models/product_product.py

```diff
@@ -27,7 +28,7 @@ def _onchange_service_fields(self):
             elif record._origin.uom_id:
                 record.uom_id = record._origin.uom_id
             else:
-                record.uom_id = self._get_default_uom_id()
+                record.uom_id = self.product_tmpl_id.default_get(['uom_id']).get('uom_id')
```

Comment:
Are you really sure this onchange is triggered with the use case you described in your commit message? 🤔 

I thought the onchange was only called inside a form view, TBH. 🤔

Reply:
Yes, if user `Create and Edit` product from SOL line it will open form view dialog or go to form view of that product from SOL
Also we want to set proper fallback when product is not service type :)